### PR TITLE
Enhanced sat list for M8N/M9N

### PR DESCRIPTION
**Pull request by TonyBlit**
Modified the sat list in order to show extra info from the following PR: https://github.com/betaflight/betaflight/pull/10921

New layout:
Screenshot 2021-09-17 at 06 36 43

New sat list in action:
Screenshot 2021-09-09 at 17 42 34

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/tabs/gps.html
Comment (haslinghuis) [inline src/tabs/gps.html:56]: Should this be removed?

File: src/js/tabs/gps.js
Comment (haslinghuis) [inline src/js/tabs/gps.js:115]:
```suggestion
                    if (FC.GPS_DATA.chn[i] >== 7) {
```

File: src/js/tabs/gps.js
Comment (haslinghuis) [inline src/js/tabs/gps.js:125]:
```suggestion
                                quality += 'no signal | ';
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]
Thanks for contributing.
Please remove commented out code.
Some remarks I made are valid in different places.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/tabs/gps.html
Comment (TonyBlit) [reply src/tabs/gps.html:56]: Actually, I forgot to ask about this. How does the localization system work? do I have to add gpsSignalGnssId with the "Gnss Id" string to the messages.json in all languages? In this case it's easy, but what about other strings that might need to be translated?

File: src/js/tabs/gps.js
Comment (TonyBlit) [reply src/js/tabs/gps.js:125]: I have reworked those ugly switch cases, so this is no longer an issue :-)

File: src/tabs/gps.html
Comment (McGiverGim) [reply src/tabs/gps.html:56]: Not, only English. We have translators that will do the rest of languages

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/tabs/gps.html
Comment (haslinghuis) [inline src/tabs/gps.html:253]: 👾 was on the right position 😆

**Review by haslinghuis — COMMENTED** [review]
Please also fix the sonar and squash your commits when done. Looks good.

**Comment by McGiverGim** [comment]
> Please also fix the sonar and squash your commits when done. Looks good.

Only as a help for @TonyBlit , not all the Sonar issues need to be fixed. Sometimes they have no sense. In this case, the literal that has been duplicated I think it does not need to be fixed, it is part of an array and will be strange to define this outside and not the other elements. 
The other two issues can be fixed.

File: src/tabs/gps.html
Comment (TonyBlit) [reply src/tabs/gps.html:253]: ouch :-)

**Comment by TonyBlit** [comment]
@McGiverGim yeah, the first one it doesn't make sense, indeed. The other ones made me think I forgot to localize the messages, so I've fixed everything and squashed, hope all is good now.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by McGiverGim — APPROVED** [review]
To me is good enough... good work! :)
As suggestion, in the Configurator PRs is good to attach a screen capture. It helps with the review.

**Review by haslinghuis — DISMISSED** [review]
LGTM

**Comment by McGiverGim** [comment]
@TonyBlit I've seen this in the firmware PR:

> gps_ublox_use_galileo will no longer disable QZSS as it's suggested on the M8N manual

Can you modify this description:

https://github.com/betaflight/betaflight-configurator/blob/dfd8a925d04042c52a8786789e8673a14a455c08/locales/en/messages.json#L1465-L1468

I think something like:

> It enables the Galileo system (European). Until Betaflight 4.2 this implied to remove the QZSS system (Japanese) too.

Sorry, I'm not English native, I'm sure you will find something better :)

**Comment by TonyBlit** [comment]
@McGiverGim Well spotted! I was not even conscious of this message.

I've been tempted to mention ublox_use_galileo is only for M8N, but given M9N has galileo already enabled, maybe it's not worth complicating the sentence until M9N is widely adopted.

**Review by haslinghuis — DISMISSED** [review]

**Review by McGiverGim — DISMISSED** [review]

**Review by ctzsnooze — DISMISSED** [review]

**Review by limonspb — CHANGES_REQUESTED** [review]
@TonyBlit please squash in 1 commit  and i think its ready to go.

**Review by limonspb — DISMISSED** [review]

**Review by haslinghuis — DISMISSED** [review]

**Review by blckmn — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> PASS

**Review by McGiverGim — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Review by ctzsnooze — DISMISSED** [review]

File: src/js/tabs/gps.js
Comment (chmelevskij) [inline src/js/tabs/gps.js:97]:
```suggestion
                const maxUIChannels = 32;
                let channels = Math.min(maxUIChannels, FC.GPS_DATA.chn.length);
```

**Review by chmelevskij — APPROVED** [review]

File: src/js/tabs/gps.js
Comment (TonyBlit) [reply src/js/tabs/gps.js:97]: good suggestion, thanks!

**Review by chmelevskij — APPROVED** [review]

**Review by ctzsnooze — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

File: src/tabs/gps.html
Comment (Asizon) [inline src/tabs/gps.html:56]: The use of styles into html part is old, now we try to move all styles in to css file.

**Review by Asizon — CHANGES_REQUESTED** [review]
Really good features, my requested changes are not to important but if you can take a look would be nice

File: src/tabs/gps.html
Comment (chmelevskij) [reply src/tabs/gps.html:56]: I wouldn't block this though. Both inline styles and styles files are redundant if we just move to vue, so one way or the other re-work will be needed.

**Review by Asizon — APPROVED** [review]

**Comment by haslinghuis** [comment]
Sonar disabled us here from merging this 😕

**Review by blckmn — APPROVED** [review]

**Pull request merged**
